### ISSUE TITLE
core/request : node provider only redirects URLs with protocol and host

### ISSUE THREAD

**Issue by sebilasse** [dojo/core#278]
"redirects" :
from looking at the code it seems that the node provider does not resolve the `nativeResponse.headers.location` against the baseUrl.

seen in the wild :
If you come from 
`https://stackoverflow.com/users{/userId}` 
you are redirected to
`/users{/userId}{/userName}`
and that _SHOULD_ resolve to

`https://stackoverflow.com/users{/userId}{/userName}` ... 

same for `//stackoverflow.com/whatever` where it should keep the protocol ...

**Comment by sebilasse** [comment]
I am using

```Typescript
import * as URL from 'url';
/* ... */
const u = URL.parse(nativeResponse.headers.location);
const redirectUrl = (!u.host || !u.protocol) ?
    URL.resolve(response.url, nativeResponse.headers.location) : nativeResponse.headers.location;
```

**Comment by rorticus** [comment]
Fix for this in https://github.com/dojo/core/pull/261

**Comment by sebilasse** [comment]
@rorticus @dylans Thanks for fixing this.
And (related to redirects) a feature request ;) : 
It could be useful to store the redirected URLs in an array in the options object. 
The options are returned as well and the user now knows the "URLs in beetween".

**Comment by dylans** [comment]
> And (related to redirects) a feature request ;) :
> It could be useful to store the redirected URLs in an array in the options object.
> The options are returned as well and the user now knows the "URLs in beetween".

Please open a new issue for this and we'll consider how to address it, otherwise it will get lost in the noise of this issue. :)

**Comment by rorticus** [comment]
Fixed in #261
